index.js

```diff
@@ -25,4 +25,14 @@ function GraphQLAdapter(options) {
   return router
 }
 
-module.exports = GraphQLAdapter
+let services
+module.exports = (srv, options) => {
+  if (!services) {
+    services = {}
+    cds.on('served', () => {
+      options.services = services
+      cds.app.use (options.endpoint, cds.middlewares.before, GraphQLAdapter(options), cds.middlewares.after)
```

Comment:
Should we separately expose `GraphQLAdapter` as an API? Otherwise a developer would need to manually call the function once for every service and rely on the `cds.on('served', ...)` event.